### PR TITLE
Add to cart adjustments

### PR DESCRIPTION
**Pull request by vileyne**
- use Event Delegation for add to cart action
- removed updateMagentoCartSectionId from AddToCart.js as not needed due to semi lightna

### CODE REVIEW

File: code/lightna/magento-os-frontend/webpack.yaml [hunk 1]
```diff
@@ -10,3 +10,5 @@ entry:
       ProductOptions: product/Options
       Gallery: product/Gallery
       Facets: listing/Facets
+    module:
+      updateMagentoCartSectionId: common/updateMagentoCartSectionId
```
Comment (vesan-83) [inline code/lightna/magento-os-frontend/webpack.yaml:14]: it seems we don't need it anymore, it was needed when we have both magento and lightna minicart on the website but now we have globally our header

File: code/lightna/magento-os-frontend/js/common/AddToCart.js [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
     }
 
     bindEvents() {
-        this.components.forEach((component) => {
-            const actionTrigger = $('[data-action="add-to-cart"]', component);
-            actionTrigger.addEventListener('click', () => {
-                this.addProduct(component, actionTrigger);
-                this.toggleAnimation(actionTrigger, true);
-            });
+        $('body').addEventListener('click', (event) => {
```
Comment (vesan-83) [inline code/lightna/magento-os-frontend/js/common/AddToCart.js:19]: I remember we discussed it, could you remind what it simplifies?

File: code/lightna/magento-os-frontend/webpack.yaml (hunk 1 above)
Comment (vileyne) [reply code/lightna/magento-os-frontend/webpack.yaml:14]: hmm... but what if we don't use semi-lightna? maybe we should remove it from common entry in semi-module? it is possible?

File: code/lightna/magento-os-frontend/js/common/AddToCart.js (hunk 2 above)
Comment (vileyne) [reply code/lightna/magento-os-frontend/js/common/AddToCart.js:19]: we do not create instances for each component; once new content lazy loaded there is no need to init add-to-cart for it/add event listener

Comment (vesan-83) [reply code/lightna/magento-os-frontend/js/common/AddToCart.js:19]: that's good, but I'm thinking if it's really scalable...

Comment (vileyne) [reply code/lightna/magento-os-frontend/js/common/AddToCart.js:19]: take your time

File: code/lightna/magento-os-frontend/webpack.yaml (hunk 1 above)
Comment (vesan-83) [reply code/lightna/magento-os-frontend/webpack.yaml:14]: it's hard to imagine such case on real project, this case means that Lightna pages and Magento pages have different headers

Comment (vileyne) [reply code/lightna/magento-os-frontend/webpack.yaml:14]: let's remove it completely than

Comment (vileyne) [reply code/lightna/magento-os-frontend/webpack.yaml:14]: I'll move into separate PR if you think too long on another thread 😃

File: code/lightna/magento-os-frontend/js/common/AddToCart.js [hunk 3]
```diff
@@ -1,71 +1,65 @@
...
+            const trigger = this.getAddToCartTrigger(event.target);
+            if (trigger) {
+                const component = trigger.closest(this.component);
+                this.onAddProductStart(trigger);
+                this.addProduct(component)
+                    .then(this.onAddProductComplete.bind(this, trigger));
```
Comment (vesan-83) [inline code/lightna/magento-os-frontend/js/common/AddToCart.js:25]: I'n not sure we have a complete here, complete triggers even if error was thrown, but it's not triggered in our case. Dunno if we need true complete event.
maybe beforeAddProduct/afterAddProduct?

Comment (vileyne) [reply code/lightna/magento-os-frontend/js/common/AddToCart.js:25]: 🫡

File: code/lightna/magento-os-frontend/js/common/AddToCart.js [hunk 4]
```diff
@@ -1,71 +1,65 @@
...
-                this.addProduct(component, actionTrigger);
-                this.toggleAnimation(actionTrigger, true);
-            });
+        $('body').addEventListener('click', (event) => {
+            const trigger = this.getAddToCartTrigger(event.target);
+            if (trigger) {
+                const component = trigger.closest(this.component);
+                this.beforeAddProduct(trigger);
+                this.addProduct(component)
+                    .then(this.afterAddProduct.bind(this, trigger));
```
Comment (vesan-83) [inline code/lightna/magento-os-frontend/js/common/AddToCart.js:25]: could we move beforeAddProduct + addProduct into onAddPropduct?

Comment (vesan-83) [reply code/lightna/magento-os-frontend/js/common/AddToCart.js]: done, please merge

**Review by vesan-83 — APPROVED** [review]

**Pull request merged**